src/physics/autompo/opsum_to_mpo.jl

```diff
@@ -1,13 +1,13 @@
-function svdMPO(os::OpSum{C}, sites; kwargs...)::MPO where {C}
+function svdMPO(
+  ValType::Union{Type{Float64},Type{ComplexF64}}, os::OpSum{C}, sites; kwargs...
```

Comment:
```suggestion
  ValType::Type, os::OpSum{C}, sites; kwargs...
```

Reply:
```console
$ julia --version
julia version 1.9.2

# ValType::Union{Type{Float64},Type{ComplexF64}}
# (https://github.com/ITensor/ITensors.jl/pull/1183/commits/c52861c9bae7709d283bd87d84eef3bad0c59951)
$ julia mpo_bench.jl
  23.923810 seconds (44.09 M allocations: 3.384 GiB, 5.20% gc time, 94.81% compilation time)
  1.142302 seconds (6.28 M allocations: 968.563 MiB, 15.38% gc time)
  1.113393 seconds (6.28 M allocations: 968.563 MiB, 13.13% gc time)
 17.854060 seconds (21.06 M allocations: 6.434 GiB, 5.61% gc time, 46.11% compilation time)
  8.215179 seconds (3.96 M allocations: 5.315 GiB, 6.19% gc time)
  8.249937 seconds (3.96 M allocations: 5.315 GiB, 4.95% gc time)

# your suggestion
# `ValType::Type`
$ julia mpo_bench.jl
 23.146701 seconds (80.69 M allocations: 4.215 GiB, 6.02% gc time, 90.06% compilation time)
  2.489611 seconds (39.05 M allocations: 1.523 GiB, 14.74% gc time)
  2.161249 seconds (39.05 M allocations: 1.523 GiB, 12.96% gc time)
 24.621732 seconds (236.28 M allocations: 10.059 GiB, 7.88% gc time, 32.82% compilation time)
 15.978803 seconds (219.25 M allocations: 8.944 GiB, 8.82% gc time)
 15.857048 seconds (219.25 M allocations: 8.944 GiB, 8.75% gc time)
```

Reply:
I'm not sure why `ValType::Type` is not enough for julia in terms of performance 🤔

Do you like `ValType::Type{<:Number}`? I get the results:

```
 julia mpo_bench.jl
 22.648911 seconds (44.11 M allocations: 3.383 GiB, 5.10% gc time, 94.79% compilation time)
  1.195179 seconds (6.30 M allocations: 967.310 MiB, 15.12% gc time)
  1.117565 seconds (6.30 M allocations: 967.310 MiB, 13.39% gc time)
 17.251445 seconds (21.06 M allocations: 6.434 GiB, 5.82% gc time, 49.08% compilation time)
  8.406711 seconds (3.96 M allocations: 5.315 GiB, 5.71% gc time)
  8.352865 seconds (3.96 M allocations: 5.315 GiB, 4.92% gc time)
```

Reply:
Wow, that's surprising to me, thanks for checking. That definitely goes against my understanding of how Julia does optimizations. I generally just think of type annotations in function signatures as a way to control method dispatch, and therefore I don't expect it to have performance implications for a given method. Perhaps there is something subtle about one type annotation versus the other forcing the Julia compiler to specialize the method more for the input type...

`ValType::Type{<:Number}` seems like a reasonable signature, I wouldn't expect non-number types to be passed anyway.